src/zabapgit_authorizations.prog.abap

```diff
@@ -0,0 +1,49 @@
+*&---------------------------------------------------------------------*
+*&  Include           ZABAPGIT_AUTHORIZATIONS
+*&---------------------------------------------------------------------*
+
+TYPES: ty_activity TYPE string.
+
+CONSTANTS: BEGIN OF gc_activities,
+             uninstall TYPE ty_activity VALUE 'UNINSTALL',
+           END OF gc_activities.
+
+INTERFACE lif_auth.
+  METHODS:
+    is_allowed
+      IMPORTING iv_activity       TYPE ty_activity
```

Comment:
I'd add optional `iv_param type string` for the case of some future extensions. E.g. repo_id can be a parameter to a certain auth rule.